api/src/main/java/io/opentelemetry/common/BooleanValuedKey.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+public interface BooleanValuedKey {
+  String key();
```

Comment:
What is the benefit of having this as an interface? Also have you thought about doing something like in the proto, extend the AttributeValue to be AttributeKeyValue?

Reply:
Yes! This was another possible suggestion in #1076 . There, we were calling it simply "Attribute". That's another possibility, but would introduce an extra allocation for each attribute application, which probably isn't desirable.